st2api/tests/unit/controllers/v1/test_executions_filters.py

```diff
@@ -303,5 +303,12 @@ def test_filters_view(self):
         response = self.app.get('/v1/executions/views/filters')
         self.assertEqual(response.status_int, 200)
         self.assertIsInstance(response.json, dict)
-        for key, value in six.iteritems(history_views.ARTIFACTS['filters']):
+        for key, value in six.iteritems(history_views.ARTIFACTS['filters']['default']):
+            self.assertEqual(set(response.json[key]), set(value))
+
+    def test_filters_view_specific_types(self):
+        response = self.app.get('/v1/executions/views/filters?types=action,user,nonexistent')
+        self.assertEqual(response.status_int, 200)
+        self.assertIsInstance(response.json, dict)
```

Comment:
To make sure that the controller indeed works as expected it would also be good to check the length of the dict (should be 2 here).